src/pages/api/datahub.ts

```diff
@@ -0,0 +1,70 @@
+import { ApiResponse, handlerWrapper } from '@/server/common'
+import {
+  CreatePostOptimisticInput,
+  UpdatePostBlockchainSyncStatusInput,
+  UpdatePostOptimisticInput,
+} from '@/server/datahub/generated-mutation'
+import {
+  createPostData,
+  notifyCreatePostFailedOrRetryStatus,
+  notifyUpdatePostFailedOrRetryStatus,
+  updatePostData,
+} from '@/server/datahub/post'
+import { datahubMutationWrapper, RateLimitError } from '@/server/datahub/utils'
+import { z } from 'zod'
+
+export type DatahubMutationInput =
+  | {
+      action: 'create-post'
+      payload: CreatePostOptimisticInput
+    }
+  | {
+      action: 'update-post'
+      payload: UpdatePostOptimisticInput
+    }
+  | {
+      action: 'notify-create-failed'
+      payload: UpdatePostBlockchainSyncStatusInput
+    }
+  | {
+      action: 'notify-update-failed'
+      payload: UpdatePostBlockchainSyncStatusInput
+    }
+export type ApiDatahubResponse = ApiResponse
+
+export default handlerWrapper({
```

Comment:
I little bit confused, that we are using backend for sending data to backend

So we just increase time to sharing data by network

Any reason?

Reply:
its request from max, so we can have token validation in datahub, where the token is only from backend side

Reply:
Oki, ok for now, but I'll talk w/ him